devito/nodes.py

```diff
@@ -473,9 +473,21 @@ def __init__(self, name, body, retval, parameters=None, prefix=('static', 'inlin
         self.name = name
         self.body = as_tuple(body)
         self.retval = retval
-        self.parameters = as_tuple(parameters)
         self.prefix = prefix
 
+        if all(isinstance(i, ArgumentProvider) for i in parameters):
```

Comment:
No, because if they are not, they would be `Argument`s and won't have this field.